tsconfig.json

```diff
@@ -0,0 +1,15 @@
+{
+  "compilerOptions": {
+    "target": "ES5",
+    "outDir": "types",
+    "newLine": "LF",
+    "allowJs": true,
+    "declaration": true,
+    "emitDeclarationOnly": true
+  },
+  "include": [
+    "index.js",
+    "src/**/*"
+  ],
+  "compileOnSave": false
+}
```

Comment:
We use this config https://github.com/webpack-contrib/terser-webpack-plugin/blob/master/tsconfig.json for all repos, let's use it?

Reply:
Feel free to feedback

Reply:
- `target` -- I'm not sure if this affects the output of type definitions. I can change it to `esnext` if you prefer. Changing it to `esnext` will change `module` to `ES6`, which will in turn change `moduleResolution` to `Classic` (which would then need to be switched back to `Node`).
- `moduleResolution` -- I'll add this if you want `target` changed. (it should ideally be proper case though [`Node`])
- `allowJs` -- this is implied by `checkJs` in TS 4.1, so you don't need to include both.
- `checkJs` -- I haven't enabled checking because there are 55 TS errors. You would need to fix those if you want it enabled.
- `strict` -- I have added this.
- `types` -- Not sure why this is needed?
- `resolveJsonModule` -- I have added this.
- `allowSyntheticDefaultImports` -- I have added this.
- `include` -- I have prefixed this with `./`.
- `outDir` -- Instead of specifying it in the `tsc` command.
- `newLine` -- This is necessary to ensure that LF is used when outputting types on Windows.
- `declaration` / `emitDeclarationOnly` -- Instead of specifying it in the `tsc` command.

Reply:
- `target` for generation latest es featurs
- `moduleResolution`, yep, we need set Node here
- `allowJs`
> Report errors in .js files. Use in conjunction with --allowJs.

So both were used, maybe I something missed, but docs say we should use both

- `types` respect `node` default types, we don't use them in this PR, but I think it is good to set
- `outDir`, `declaration` and `emitDeclarationOnly`, we have two steps: linting and generation, better to have separate command for this

- agree about `newLine`

Reply:
- `allowJs` -- The docs haven't been updated. See the [release announcement](https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/#checkjs-implies-allowjs).
- `checkJs` -- I can't replace `allowJs` with `checkJs` unless you fix the 55 errors. 🙂
  - Please let me know what you want to do here.
- `outDir`, `declaration` and `emitDeclarationOnly` -- So you want me to move these to the command?, like:

`"build:types": "tsc --declaration --emitDeclarationOnly --outDir types && prettier \"types/**/*.ts\" --write"`

Reply:
> allowJs -- The docs haven't been updated. See the release announcement.

Thanks for link :+1: 

> checkJs -- I can't replace allowJs with checkJs unless you fix the 55 errors. 

Yep, we will do it latest

> outDir, declaration and emitDeclarationOnly -- So you want me to move these to the command?, like:

https://github.com/webpack/schema-utils/blob/master/package.json#L23

Reply:
Do you want the name of the folder to be changed from `types` to `declarations`?

Reply:
I don't have strong opinion here, but let's name `types`